src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -338,8 +340,12 @@ private static SyntaxNode WithBlockBody(SyntaxNode node, BlockSyntax body)
 
             var firstStatementAffectedIndex = GetFirstStatementAffectedIndex(innermostCommonBlock, matches, GetStatements(innermostCommonBlock).IndexOf(allAffectedStatements.Contains));
 
+            // we already filtered the matches based on rvalue/lvalue and includeLValues, 
+            // and while theoretically we could pass the includeLValues as is to the rewrite function,
+            // this will cause it to refilter the list of nodes, and that will fail because they've been rewritten above
+            // and now are out of sync with the semantic model. I could rebuild the semantic model, but really, our work here is done
```

Comment:
sorry can you give an example of what yuo mean.  this comment doesn't really make it clear waht hte issue is here.